tree/dataframe/test/dataframe_simple.cxx

```diff
@@ -48,29 +48,36 @@ class RDFSimpleTests : public ::testing::TestWithParam<bool> {
 // Create file `filename` containing a test tree `treeName` with `nevents` events
 void FillTree(const char *filename, const char *treeName, int nevents = 0)
 {
-   TFile f(filename, "RECREATE");
-   TTree t(treeName, treeName);
-   t.SetAutoFlush(1); // yes, one event per cluster: to make MT more meaningful
+   // NOTE(gparolini): these TFile and TTree are created on the heap to work around a know bug that can
+   // cause a TObject to be incorrectly marked as "on heap" and attempted to be freed despite
+   // living on the stack.
+   // The bug is caused by the magic bit pattern `kObjectAllocMemValue` used by TStorage to
+   // mark a heap object appearing by chance on the stack.
+   // This is not a problem with a clear solution and in fact the whole heap detection system relies on UB,
+   // so for now we are forced to work around the bug rather than fixing it.
+   auto t = std::make_unique<TTree>(treeName, treeName);
+   auto f = std::make_unique<TFile>(filename, "RECREATE");
```

Comment:
The file should be created before the `TTree` so that it can attach to the file as intended. With this code, the data is kept all in memory until `Write` is called.  This also most likely disable the meaning of `SetAutoFlush(1)` as the `TTree` can not flush to disk since it is not attached.  Use either:
```
   auto t = std::make_unique<TTree>(treeName, treeName);
   auto f = std::make_unique<TFile>(filename, "RECREATE");
   t->SetDirectory(f);
```
or
```
   auto f = std::make_unique<TFile>(filename, "RECREATE");
   auto t = std::make_unique<TTree>(treeName, treeName);
```

Reply:
The reason why I inverted the declarations was because the test would crash otherwise (on delete of one of the objects). I thought the reason was the order of declarations, but I guess not at this point. I'll investigate more on what's happening.

Reply:
So I guess TFile::Close() deletes the TTree and the best solution is just remove the explicit call to `t->Close()`, which seems to fix the issue.

Reply:
#16421

Reply:
> TFile::Close() deletes the TTree

Yes, it does.  And we recommend not to call it.

Reply:
The only 'reason' to use `Close` is if one wants to re-use the `TFile` object at a later point.